### PR TITLE
[CPU] Reduce node migration on nGraph

### PR DESCRIPTION
**Pull request by mandrono**

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp [hunk 1]
```diff
@@ -1319,51 +1320,106 @@ struct jit_uni_reduce_post_kernel_f32 : public jit_uni_reduce_post_kernel, publi
...
+    } else {
+        THROW_IE_EXCEPTION << "Reduce node with name " << op->get_friendly_name()
+                           << " is incompatibility with ArithmeticReductionKeepDims and LogicalReductionKeepDims";
+    }
+}
+
+std::map<const ngraph::DiscreteTypeInfo, std::function<void(const std::shared_ptr<ngraph::Node>&, MKLDNNReduceNode&)>> MKLDNNReduceNode::initializers = {
+    {ngraph::opset4::ReduceL1::type_info, [](const std::shared_ptr<ngraph::Node>& op, MKLDNNReduceNode& node) {
+        node.algorithm = ReduceL1;
+        node.keep_dims = getKeepDims(op);
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp:1337]: It is used equally in each initializer. I think we should: 
1) Check that `op` is derived from `ArithmeticReductionKeepDims` or `LogicalReductionKeepDims` in `isSupportedOperation`
2) Move `getKeepsDims` implementation inside the node ctor.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp [hunk 2]
```diff
@@ -1319,51 +1320,106 @@ struct jit_uni_reduce_post_kernel_f32 : public jit_uni_reduce_post_kernel, publi
...
 
-MKLDNNReduceNode::MKLDNNReduceNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
-        : MKLDNNNode(layer, eng, cache) {}
+static bool getKeepDims(const std::shared_ptr<ngraph::Node>& op) {
+    if (const auto reduce = std::dynamic_pointer_cast<ngraph::op::util::ArithmeticReductionKeepDims>(op)) {
+        return reduce->get_keep_dims();
+    } else if (const auto reduce = std::dynamic_pointer_cast<ngraph::op::util::LogicalReductionKeepDims>(op)) {
+        return reduce->get_keep_dims();
+    } else {
+        THROW_IE_EXCEPTION << "Reduce node with name " << op->get_friendly_name()
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp:1329]: `is incompatibility with ...` -> `is not derived from ...`

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp (hunk 1 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp]: done

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp (hunk 2 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp]: fixed

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**